neurodamus/ngv.py

```diff
@@ -19,19 +19,136 @@
+
+    def append(self, item):
+        self._body.append(item)
+
+    def pop(self, index=-2):
+        index = self._sanitize_index(index)
+        if index == len(self) - 1:
+            raise IndexError("GlutList tail cannot be popped")
+        self._body.pop(index)
+
+    def __setitem__(self, index, value):
+        index = self._sanitize_index(index)
+        if index == len(self) - 1:
+            self._tail = value
+        else:
+            self._body.__setitem__(index, value)
+
+    def __iter__(self):
+        yield from self._body
+        yield self._tail
+
+    def to_list(self):
+        return [*self._body, self._tail]
+
+    def __eq__(self, other):
+        if isinstance(other, GlutList):
+            return self._body == other._body and self._tail == other._tail
+        if isinstance(other, list):
+            return self.to_list() == other
+        return NotImplemented
+
+    @property
+    def tail(self):
+        return self._tail
+
+    @tail.setter
+    def tail(self, value):
+        self._tail = value
+
+    def __str__(self):
+        return str(self.to_list())
+
+
 class Astrocyte(BaseCell):
-    __slots__ = ("_glut_list", "_nseg_warning", "_secidx2names")
+    __slots__ = ("_glut_list", "_nseg_warning", "section_names")
 
     def __init__(self, gid, meinfos, circuit_conf):
-        """Instantiate a new Cell from node info."""
+        """Initialize an Astrocyte cell.
+
+        - Load and attach morphology to the cell.
+        - Insert a GlutReceive point process in every section.
+        - Insert the cadifus mechanism in every section.
+        - Connect GlutReceive to cadifus using a pointer.
+        - Insert a GlutReceiveSoma in the soma (used only for recording).
+        """
         super().__init__(gid, meinfos, None)
-        morpho_path = circuit_conf.MorphologyPath
-        morph_filename = meinfos.morph_name + "." + circuit_conf.MorphologyType
-        morph_file = os.path.join(morpho_path, morph_filename)
-        self._cellref, self._glut_list, self._secidx2names, self._nseg_warning = self._init_cell(
-            gid, morph_file
+
+        # Compose the path to the morphology file
+        morph_file = (
+            Path(circuit_conf.MorphologyPath)
+            / f"{meinfos.morph_name}.{circuit_conf.MorphologyType}"
         )
+
+        # Create the cell and load morphology
+        self._cellref = Nd.Cell(gid)
+        morph = MorphIOWrapper(morph_file)
+        self._cellref.AddHocMorph(morph.morph_as_hoc())
+
+        self.glut_list = GlutList()
+        self._nseg_warning = 0
+
+        # Recalculate number of segments and sections
+        self._cellref.geom_nseg_fixed()
+        self._cellref.geom_nsec()
+
+        logging.debug("Instantiating NGV cell gid=%d", gid)
+
+        # Insert mechanisms and glutamate receptors in each section
+        for sec in self._cellref.all:
+            if sec.nseg > 1:
+                self._nseg_warning = 1
```

Comment:
what sort of warning is this? it also seems to be used by AstrocyteManager, so should it be public?

Reply:
there is an issue on this #261. I will takle this once I proceed on AstrocyteManager

Reply:
ok